cpp/tests/centrality/edge_betweenness_centrality_test.cu

```diff
@@ -0,0 +1,334 @@
+  std::vector<double> deltas(number_of_vertices);
+
+  std::vector<VT> neighbors;
+
+  if (sources) {
+    for (VT source_idx = 0; source_idx < number_of_sources; ++source_idx) {
+      VT s = sources[source_idx];
+      // Step 1: Single-source shortest-paths problem
+      //   a. Initialization
+      ref_bfs<VT, ET>(indices, offsets, number_of_vertices, Q, S, dist, pred, sigmas, s);
+      //  Step 2: Accumulation
+      //          Back propagation of dependencies
+      ref_accumulation<VT, ET, WT, result_t>(
+        result, indices, offsets, number_of_vertices, S, pred, sigmas, deltas, s);
+    }
+  } else {
+    for (VT s = 0; s < number_of_vertices; ++s) {
+      // Step 1: Single-source shortest-paths problem
+      //   a. Initialization
+      ref_bfs<VT, ET>(indices, offsets, number_of_vertices, Q, S, dist, pred, sigmas, s);
+      //  Step 2: Accumulation
+      //          Back propagation of dependencies
+      ref_accumulation<VT, ET, WT, result_t>(
+        result, indices, offsets, number_of_vertices, S, pred, sigmas, deltas, s);
+    }
+  }
+}
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void reference_rescale(result_t *result,
+                       bool directed,
+                       bool normalize,
+                       VT const number_of_vertices,
+                       ET const number_of_edges)
+{
+  result_t rescale_factor            = static_cast<result_t>(1);
+  result_t casted_number_of_vertices = static_cast<result_t>(number_of_vertices);
+  if (normalize) {
+    if (number_of_vertices > 1) {
+      rescale_factor /= ((casted_number_of_vertices) * (casted_number_of_vertices - 1));
+    }
+  } else {
+    if (!directed) { rescale_factor /= static_cast<result_t>(2); }
+  }
+  for (auto idx = 0; idx < number_of_edges; ++idx) { result[idx] *= rescale_factor; }
+}
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void reference_edge_betweenness_centrality(
+  cugraph::experimental::GraphCSRView<VT, ET, WT> const &graph,
+  result_t *result,
+  bool normalize,
+  VT const number_of_sources,
+  VT const *sources)
+{
+  VT number_of_vertices = graph.number_of_vertices;
+  ET number_of_edges    = graph.number_of_edges;
+  thrust::host_vector<VT> h_indices(number_of_edges);
+  thrust::host_vector<ET> h_offsets(number_of_vertices + 1);
+
+  thrust::device_ptr<VT> d_indices((VT *)&graph.indices[0]);
+  thrust::device_ptr<ET> d_offsets((ET *)&graph.offsets[0]);
+
+  thrust::copy(d_indices, d_indices + number_of_edges, h_indices.begin());
+  thrust::copy(d_offsets, d_offsets + (number_of_vertices + 1), h_offsets.begin());
+
+  cudaDeviceSynchronize();
+
+  reference_edge_betweenness_centrality_impl<VT, ET, WT, result_t>(
+    &h_indices[0], &h_offsets[0], number_of_vertices, result, sources, number_of_sources);
+  reference_rescale<VT, ET, WT, result_t>(
+    result, graph.prop.directed, normalize, number_of_vertices, number_of_edges);
+}
+// Explicit declaration
```

Comment:
This is not a declaration, "explicit instantiation" is a proper terminology.

Reply:
Updated in 257b2f9.